### PR TITLE
Added Highest Score Feature in Snake Game

### PR DESCRIPTION
**Pull request by Pratik2803**
Hello Sunandan,

I have added highest score feature in current snake game. Kindly review it as per your earliest convenience...

Best Regards,
Pratik

### CODE REVIEW

File: Day_20/snake.py [hunk 1]
```diff
@@ -15,7 +15,6 @@ def __init__(self):
         self.create_snake()
         self.head = self.segments[0]
 
-
```
Comment (snaps2k) [inline Day_20/snake.py:18]: Blank link added, ideally this should be part of commit

File: Day_20/scoreboard.py [hunk 2]
```diff
@@ -14,18 +14,28 @@ def __init__(self):
...
+        self.current_score = 0
+        self.high_score = 0
         self.goto(x=SCOREBOARD_XCOR, y=SCOREBOARD_YCOR)
+
+        self.get_highest_score()
         self.scoreboard_text()
 
+    def get_highest_score(self):
+        with open(file='./highest_score.txt', mode='r', encoding='utf-8') as f:
+            self.high_score = f.read()
```
Comment (snaps2k) [inline Day_20/scoreboard.py:26]: high score should be int when it is loaded, as the value can be used at many places that will lead to multiple type casting. like below where you are comparing it.

File: Day_20/scoreboard.py [hunk 3]
```diff
@@ -14,18 +14,28 @@ def __init__(self):
...
     def scoreboard_text(self):
         self.clear()
-        self.write(f"Score: {self.counter}", align=ALIGNMENT, font=FONT)
+        self.write(f"Score: {self.current_score}  High Score: {self.high_score}", align=ALIGNMENT, font=FONT)
 
-    def game_over(self):
-        self.goto(0,0)
-        self.write(f"Game Over.", align=ALIGNMENT, font=FONT)
+    def scoreboard_reset(self):
+        if int(self.high_score) < self.current_score:
```
Comment (snaps2k) [inline Day_20/scoreboard.py:33]: correct this too when previous comment is resolved.

File: Day_20/scoreboard.py (hunk 2 above)
Comment (Pratik2803) [reply Day_20/scoreboard.py:26]: Agreed. Raising a new pull request with correction

File: Day_20/scoreboard.py (hunk 3 above)
Comment (Pratik2803) [reply Day_20/scoreboard.py:33]: Agreed raising a new pull request with correction

**Pull request merged**